File: services/src/main/java/com/teamdev/filehub/views/userprofile/UserProfile.java
package com.teamdev.filehub.views.userprofile;

/**
 * Server response which is intended to store
 * information about the user profile.
 */
public class UserProfile {

    private final String login;
    private final String rootFolderId;

    public UserProfile(String login, String rootFolderId) {
        this.login = login;
        this.rootFolderId = rootFolderId;
    }

    public String username() {
        return login;
    }

    public String rootFolderId() {
        return rootFolderId;
    }
}

File: web/src/main/js-app/src/state-management/user/user-profile.js
/**
 * User profile.
 */
export class UserProfile {
  #username;
  #rootFolderId;

  /**
   * @param {string} username
   * @param {string} rootFolderId
   */
  constructor(username, rootFolderId) {
    this.#username = username;
    this.#rootFolderId = rootFolderId;
  }

  /**
   * @returns {string}
   */
  get username() {
    return this.#username;
  }

  /**
   * @returns {string}
   */
  get rootFolderId() {
    return this.#rootFolderId;
  }
}
